src/steps/reservationsSteps.ts

```diff
@@ -0,0 +1,198 @@
+import { Given, When, Then } from "@cucumber/cucumber";
+
+import { ICustomWorld } from "../world/customWorld";
+import { ReservationsPage } from "../pages/reservationsPage";
+import { StringUtils } from "../suport/utils/stringUtils";
+import { rateOverrideData, scheduleData } from "../suport/types/reservation.type";
+import { expect } from "@playwright/test";
+
+let reservationsPage: ReservationsPage;
+let minutes: number;
+let scheduleData1: scheduleData;
+let scheduleData2: scheduleData;
+let rateOverideObj: rateOverrideData
+
+// ----- edit No-Shoes Section
+Given(
+  "that the user is on the reservation page",
+  async function (this: ICustomWorld) {
+    reservationsPage = await this.pagesObj.reservationsPage;
+    minutes = Math.floor(Math.random() * 60) + 1;
+    scheduleData1 = StringUtils.generateRandomReservationData();
+    scheduleData2 = StringUtils.generateRandomReservationData()
+    rateOverideObj = {overrideName: scheduleData1.overrideName, amount: scheduleData1.overrideAmount}
+
+    await reservationsPage.navigateToSettingsPage();
+    await reservationsPage.navigateToReservationsPage();
+  }
+);
+
+When("the user clicks on the {string} checkbox", async function (string) {
+  await reservationsPage.checkTheCheckBox(string);
+});
+
+When("the user adjusts the time duration in the checkbox", async function () {
+  await reservationsPage.insertsNoShowOffsetMinutes(minutes.toString());
+});
+
+Then("the modified data should be saved", async function () {
+  await reservationsPage.checkIfInputUpdated(scheduleData1.randomTwoDigitNumber.toString());
+});
+
+//////////////////////////////////////////
+//
+// Add, Edit, Delete a Schedule
+
+Then(
+  "the user clicks on the {string} button within the {string} section",
+  async function (string, string2) {
+    await reservationsPage.clickOnNewScheduleModal();
+  }
+);
+
+When("the user fills in the required “Name” input", async function () {
```

Comment:
i would recommend not to use "" for step definitions without parameters.

Reply:
done. Correct all places in the code where i used "quotation mark" without parameters.